scripts/properties2json.js

```diff
@@ -62,30 +71,37 @@ function readPropertiesFile(filePath) {
   });
 }
 
-function getContentMessages(locale) {
-  return new Promise(function(resolve, reject) {
-    FS.listTree(path.join(process.cwd(), config.src, locale), function(filePath, stat) {
+function getContentMessages(src) {
+  return function(locale) {
+    return FS.listTree(path.join(process.cwd(), src, locale), function(filePath, stat) {
       return path.extname(filePath) === ".properties";
     }).then(function(files) {
-      Promise.all(files.map(readPropertiesFile)).then(function(properties) {
+      return Promise.all(files.map(readPropertiesFile)).then(function(properties) {
         var merged_properties = {};
         properties.forEach(function(messages) {
           Hoek.merge(merged_properties, messages);
         });
-        resolve({content: merged_properties, locale: locale});
+        return {content: merged_properties, locale: locale};
       });
-    }).catch(function(e) {
-      console.log(e);
-      reject(e);
     });
-  });
+  };
 }
 
-function processMessageFiles(locales) {
-  return Promise.all(locales.map(getContentMessages));
+function applyConfig(config) {
+  getListLocales(config.src).then(function(locales) {
+    return Promise.all(locales.map(getContentMessages(config.src)));
+  }).then(function(entries) {
+    writeFile(buildDictionary(entries), config.dest.dir, config.dest.file);
+  }).catch(function(err) {
+    console.error(err);
+  });
 }
 
-getListLocales().then(processMessageFiles)
-.then(writeFile).catch(function(err) {
-  console.error(err);
+applyConfig(defaultConfig);
+applyConfig({
+  "dest": {
+    "dir": "public",
+    "file": "buyers-guide-locales.json"
+  },
+  "src": "buyers-guide-locales"
```

Comment:
The above few lines are the main point. Anything in this file that was like, config.something we a problem, now it's not.